osf/models/notable_domain.py

```diff
@@ -40,6 +45,56 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
+    @staticmethod
+    def check_resource_for_domains(obj, confirm_spam=False, send_to_moderation=False):
+        fields_to_query = []
+        for field in list(obj.SPAM_CHECK_FIELDS):
+            fields_to_query += [models.Q(**{f'{field}__regex': settings.DOMAIN_REGEX})]
+        query = reduce(operator.or_, fields_to_query)
+        has_domain = obj.__class__.objects.filter(query).filter(id=obj.id).exists()
+
+        if not has_domain:
+            return False
+        elif confirm_spam and NotableDomain.has_spam_domain(obj):
+            obj.confirm_spam(save=True)
+        elif send_to_moderation:
```

Comment:
Do we really need the `send_to_moderation` flag? If so this is more reason why this should be on the `SpamMixin` rather than the `NotableDomain` class.